docs/reference/operations/create_accounts.md

```diff
@@ -0,0 +1,112 @@
+batch, which is not legal. (`flags.linked` indicates that the chain continues to the next
+operation).
+
+### `reserved_flag`
+The account was not created.
+[`Account.flags.reserved`](../accounts.md#flagsreserved) is nonzero, but must be zero.
+
+### `reserved_field`
+The account was not created.
+[`Account.reserved`](../accounts.md#reserved) is nonzero, but must be zero.
+
+### `id_must_not_be_zero`
+The account was not created.
+[`Account.id`](../accounts.md#id) is zero, which is a reserved value.
+
+### `id_must_not_be_int_max`
+The account was not created.
+[`Account.id`](../accounts.md#id) is `2^128 - 1`, which is a reserved value.
+
+### `ledger_must_not_be_zero`
+The account was not created.
+[`Account.ledger`](../accounts.md#ledger) is zero, but must be nonzero.
+
+### `code_must_not_be_zero`
+The account was not created.
+[`Account.code`](../accounts.md#code) is zero, but must be nonzero.
+
+### `debits_pending_must_be_zero`
+The account was not created.
+[`Account.debits_pending`](../accounts.md#debits_pending) is nonzero, but must be zero.
+
+An account's debits and credits are only modified by transfers.
+
+### `debits_posted_must_be_zero`
+The account was not created.
+[`Account.debits_posted`](../accounts.md#debits_posted) is nonzero, but must be zero.
+
+An account's debits and credits are only modified by transfers.
+
+### `credits_pending_must_be_zero`
+The account was not created.
+[`Account.credits_pending`](../accounts.md#credits_pending) is nonzero, but must be zero.
+
+An account's debits and credits are only modified by transfers.
+
+### `credits_posted_must_be_zero`
+The account was not created.
+[`Account.credits_posted`](../accounts.md#credits_posted) is nonzero, but must be zero.
+
+An account's debits and credits are only modified by transfers.
+
+### `mutually_exclusive_flags`
+The account was not created.
+An account cannot be created with the specified combination of
+[`Account.flags`](../accounts.md#flags).
+
+The following flags are mutually exclusive:
+- [`Account.flags.debits_must_not_exceed_credits`](../accounts.md#flagsdebits_must_not_exceed_credits)
+- [`Account.flags.credits_must_not_exceed_debits`](../accounts.md#flagscredits_must_not_exceed_debits)
+
+### `exists_with_different_flags`
+The account already exists, but with different [`flags`](../accounts.md#flags).
+
+### `exists_with_different_user_data`
+The account already exists, but with different [`user_data`](../accounts.md#user_data).
+
+### `exists_with_different_ledger`
+The account already exists, but with different [`ledger`](../accounts.md#ledger).
+
+### `exists_with_different_code`
+The account already exists, but with different [`code`](../accounts.md#code).
+
+### `exists`
+The account already exists.
+
+With the possible exception of the following fields, the existing account is identical to the
+account in the request:
+- `debits_pending`
+- `debits_posted`
+- `credits_pending`
+- `credits_posted`
```

Comment:
Isn't it only `id` that must be identical? You don't need all other fields to be equal for us to register it as existing right?

Reply:
There are other variants of the `exists` error: `exists_with_different_ledger`, `exists_with_different_code`, etc. So if `ledger`/`code` differ, those respective errors are returned instead of `exists`.

`{debits,credits}_{pending,posted}` are a special case because clients shouldn't be creating accounts with nonzero balances anyway — that is a separate (higher-priority) error, because it implements a bug in the application or client code.

Reply:
I'm still confused though, do we try to say that things are equal if fields other than id are equal? Isn't id the only important thing in discovering existence/not?

Reply:
For all `exists_...` variants the `id` matches — that's a prerequisite.
We return a more specific error when certain fields (e.g. `ledger`) are mismatched, because that probably indicates a programmer error.

Just plain `exists` doesn't (necessarily) indicate a programmer error though — it is normal when the application is recovering from a crash and replaying operations because it isn't sure where it left off.

I'll rephrase this a bit to make that more clear.